tests/test_regression.py

```diff
@@ -0,0 +1,124 @@
+    return grids
+
+
+class _FakePDF:
+    """A Fake lhapdf-like PDF to evolve the grids"""
+
+    def __init__(self):
+        pids = np.arange(-6, 7, dtype=int)
+        pids[6] = 21
+
+        alphas = np.linspace(1.2, 1.8, len(pids))
+        betas = np.linspace(1.2, 3.8, len(pids))
+
+        self._alphas = dict(zip(pids, alphas))
+        self._betas = dict(zip(pids, betas))
+
+    def xfxQ2(self, pid, x, q2):
+        """Compute x^alpha*(1-x)^beta"""
+        alpha = self._alphas[pid]
+        beta = self._betas[pid]
+        return np.power(x, alpha) * np.power(1 - x, beta)
+
+
+def _trim_template(template_card, take_points=10):
+    """Trim the template card so that the number of x-values to compute is much smaller"""
+    card_info = safe_load(template_card.open("r", encoding="utf-8"))
+    original_x = np.array(card_info["rotations"]["xgrid"])
+    skip = int(len(original_x) / take_points)
+    idxs = list(range(0, len(original_x), skip))
+    card_info["rotations"]["xgrid"] = original_x[idxs].tolist()
+    dump(card_info, template_card.open("w", encoding="utf-8"))
+
+
+@pytest.mark.parametrize("dataset", ["LHCBWZMU8TEV", "INTEGXT3"])
+def test_regression(tmp_path, dataset, rebuild=False):
+    """Run pineko through subprocess to ensure that the shell scripts are working exactly
+    as intended.
+    """
+    # We start by downloading pineko.toml in order to generate the folder structure
+    _download_resources("pineko.toml", tmp_path)
+    # Which we create... now!
+    run(["pineko", "scaffold", "new"], cwd=tmp_path, check=True)
+
+    # Now download other necessary objects
+    _download_resources(f"data/theory_cards/{THEORYID}.yaml", tmp_path)
+    template_card = _download_resources(
+        f"data/operator_cards/{THEORYID}/_template.yaml", tmp_path
+    )
+    _trim_template(template_card)
+
+    # And use some small (but not trivial!) dataset to test
+    gridnames = _download_dataset(dataset, THEORYID, tmp_path)
+
+    # Now go, first with eko creation
+    run(
+        ["pineko", "theory", "opcards", str(THEORYID), dataset],
+        cwd=tmp_path,
+        check=True,
+    )
+    run(["pineko", "theory", "ekos", str(THEORYID), dataset], cwd=tmp_path, check=True)
+
+    # Then FK Table production!
+    run(["pineko", "theory", "fks", str(THEORYID), dataset], cwd=tmp_path, check=True)
+
+    # Now loop over the grids and check the results of the convolution with the PDF
+    pdf = _FakePDF()
+    regression_path = Path(__file__).parent / "regression_data" / f"{dataset}.npy"
+
+    result = []
+    for grid_name in gridnames:
+        fkt = FkTable.read(
+            tmp_path / "data" / "fktables" / str(THEORYID) / f"{grid_name}.pineappl.lz4"
+        )
+        result.append(fkt.convolute_with_one(2212, pdf.xfxQ2))
+    result = np.concatenate(result)
+
+    if rebuild:
+        np.save(regression_path, result)
+
+    regression_data = np.load(regression_path)
```

Comment:
I see, but then I'd split the content of the test in a separate function, instead of using the test itself to create new metadata, and here stick to computed data.

I fear that the argument is non-standard, and if you forget that default toggled, we might inadvertently run meaningless regressions, until we will find out.